/*
 * @begin:license
 *
 * Copyright (c) 2015-2019, Benjamin Niemann <pink@odahoda.de>
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along
 * with this program; if not, write to the Free Software Foundation, Inc.,
 * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 *
 * @end:license
 */

syntax = "proto2";

import "noisicaa/audioproc/public/processor_message.proto";
import "noisicaa/builtin_nodes/control_track/processor_messages.proto";
import "noisicaa/builtin_nodes/sample_track/processor_messages.proto";
import "noisicaa/builtin_nodes/instrument/processor_messages.proto";
import "noisicaa/builtin_nodes/pianoroll/processor_messages.proto";
import "noisicaa/builtin_nodes/custom_csound/processor_messages.proto";

package noisicaa.pb;

extend ProcessorMessage {
  // Control track (403xxx)
  optional CVGeneratorAddControlPoint cvgenerator_add_control_point = 403000;
  optional CVGeneratorRemoveControlPoint cvgenerator_remove_control_point = 403001;

  // Sample track (404xxx)
  optional SampleScriptAddSample sample_script_add_sample = 404000;
  optional SampleScriptRemoveSample sample_script_remove_sample = 404001;

  // Instrument (405xxx)
  optional ChangeInstrument change_instrument = 405000;

  // Pianoroll (406xxx)
  optional PianoRollAddInterval pianoroll_add_interval = 406000;
  optional PianoRollRemoveInterval pianoroll_remove_interval = 406001;

  // Custom CSound (407xxx)
  optional CustomCSoundSetScript custom_csound_set_script = 407000;
}
